be/src/runtime/fragment_mgr.cpp

```diff
@@ -881,178 +659,89 @@ Status FragmentMgr::exec_plan_fragment(const TPipelineFragmentParams& params,
 
     std::shared_ptr<QueryContext> query_ctx;
     RETURN_IF_ERROR(_get_query_ctx(params, params.query_id, true, query_ctx));
+    auto pre_and_submit = [&](int i) {
+        const auto& local_params = params.local_params[i];
+
+        const TUniqueId& fragment_instance_id = local_params.fragment_instance_id;
+        {
+            std::lock_guard<std::mutex> lock(_lock);
+            auto iter = _pipeline_map.find(fragment_instance_id);
+            if (iter != _pipeline_map.end()) {
+                // Duplicated
+                return Status::OK();
+            }
+            query_ctx->fragment_ids.push_back(fragment_instance_id);
+        }
+        START_AND_SCOPE_SPAN(tracer, span, "exec_instance");
+        span->SetAttribute("instance_id", print_id(fragment_instance_id));
 
-    const bool enable_pipeline_x = params.query_options.__isset.enable_pipeline_x_engine &&
-                                   params.query_options.enable_pipeline_x_engine;
-    if (enable_pipeline_x) {
         int64_t duration_ns = 0;
+        if (!params.__isset.need_wait_execution_trigger || !params.need_wait_execution_trigger) {
+            query_ctx->set_ready_to_execute_only();
```

Comment:
The clause should be move outside of the pre_and_submit lambda function. Since it should be called only once for a query.